src/ORM/EagerLoadedList.php

```diff
@@ -503,6 +505,32 @@ public function filterAny(...$args): static
         return $list;
     }
 
+    /**
+     * Return a copy of this list which only includes items where $fieldToFilterBy matches values in $fieldInOtherList from $list.
+     *
+     * If both fields are ID, the $list dataclass must be in the same class hierarchy as the dataclass in this list
+     *
+     * @param SS_List<DataObject> $list
+     * @param string $fieldToFilterBy The name of the field in $this to filter by
+     * @param string $fieldInOtherList The name of the field in $list to get filter values from
+     * @return static<T>
+     * @throws InvalidArgumentException
+     */
+    public function filterByList(SS_List $list, string $fieldToFilterBy = 'ID', string $fieldInOtherList = 'ID'): static
```

Comment:
```suggestion
    public function filterByList(SS_List $list, string $fieldToFilterBy = 'ID', string $fieldFromOtherList = 'ID'): static
```
Also update its uses

Reply:
Done